PrestaShop/PrestaShop
# Improve translation test module

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | Read below
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | Read below

## What's this about

This PR replaces #13845 and solves the two first problems noted previously:

1. The test module `translation_test` has been renamed to `translationtest`, since translations cannot work on modules with underscores in their name. Indeed, the domain name for `translation_test` would be `Modules.Translationtest`, which can never be reversed back to `translation_test` to find the original module. [This required updating TranslationToolsBundle](https://github.com/PrestaShop/TranslationToolsBundle/pull/64).
2. The third component of a translation domain, in the case of modules, is the original file name. Many file names contain underscore, but it wasn't supported because the translation interface in BO used Inflector::tableize to separate the domain parts (yes domains are stored without dots in the translator dictionary, but that's another story). This has been fixed in [this commit](https://github.com/PrestaShop/PrestaShop/commit/9f1c2fe9f070a8952eff1bea8eb6ebbc229954d5)

In addition, this change also allows Module AdminControllers to reference admin templates (ie. templates located in `mymodule/views/templates/admin/`) without using the full route to the template ([see change](https://github.com/PrestaShop/PrestaShop/commit/504bbd558f895d6fb9a8d362afde5ac4f3a7112c)).

## How to test

1. Copy the `translationtest` module located in `tests/Resources/modules` over to the `/modules` folder. 
1. Go to modules catalog, search "translation" and install it.
1. Configure the module, you should see a simple page containing the wording "This wording belongs to the module file".
1. Go to `/admin-dev/index.php/modules/translations`. If your shop is in French you should get a screen with two wordings "Contrôleur moderne" and "Bonjour le monde"
1. Go to `/admin-dev/index.php?controller=AdminTranslationtestFoo`. If your shop is in French you should get a page containing the wording "Le template Smarty"
1. Go to `index.php?fc=module&module=translationtest&controller=bar`. If your shop is in French you should get a blank page containing only the wording "Ceci est un contrôleur frontal"


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/15017)
